specs-go/v1/config.go

```diff
@@ -75,6 +75,7 @@ type History struct {
 }
 
 // Image is the JSON structure which describes some basic information about the image.
+// This provides the `application/vnd.oci.image.config.v1+json` mediatype when marshalled to JSON.
```

Comment:
> The mediatype is as necessary as a document providing the version or schema that it represents.

Agreed.  But if you know the media type *from the descriptor*, why repeat it in the blob itself?  The only case I've seen is for outside-of-CAS reference like [verification][1], where the caller presumably knows the media type they're trying to verify but doesn't feel like typing it out.  I'm ok with providing a self-describing `mediaType` to save them that effort, but don't feel like we have to provide it.

[1]: https://github.com/opencontainers/image-spec/pull/411#issuecomment-257489701